public/src/managers/ServerProvider.ts

```diff
@@ -233,7 +235,7 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
 
     public async getCourseLabs(courseID: number, groupLabs: boolean): Promise<IStudentLabsForCourse[]> {
         const result = await this.grpcHelper.getCourseLabSubmissions(courseID, groupLabs);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
```

Comment:
This pattern becomes:
```typescript
if (!this.responseSuccess(result)) {
    return [];
}
...
```

Reply:
PS: These OR checks are probably not exactly correct, because they require that both the success code is non-zero and that the `result.data` to be null at the same time. I feels more correct that if the success code is non-zero (failure), then the condition should "fail", and that if the success was a "success" then if there is no data, it is also a fail. But I may be wrong. And it may not matter as well...